lnclient/lnd/lnd.go

```diff
@@ -445,7 +458,7 @@ func NewLNDService(ctx context.Context, eventPublisher events.EventPublisher, ln
 					payment, err := paymentStream.Recv()
 					if err != nil {
 						logger.Logger.WithError(err).Error("Failed to receive payment")
-						continue
+						break paymentsLoop
```

Comment:
this double for loop is a bit scary and I don't think this `lndCtx.Done()` is hit if LND is not running (because this for loop will probably never be entered?)

@im-adithya this is a case with complicated logic - can you please write down the tests you did to cover these cases? Is there a simpler way we can implement it? (I'm not sure there is)

Reply:
> I don't think this lndCtx.Done() is hit if LND is not running

Yes, this won't be hit because `lndClient.SubscribePayments` call itself would fail

Reply:
I made some changes, this should be better and will also end the goroutine in all cases so it should be safer too.

Tests done to check subscription logic:

1. Normal run to see if subscription receives event and handles it properly
2. Started Hub (with node running in Polar) and then killed context from Hub's go terminal -> Goes to the <-lndCtx.Done() case instead of breaking the paymentsLoop because ctx is ended, thereby ending the goroutine
3. Started Hub (with node running in Polar) and stopped the node in Polar after the subscription is started -> Breaks the payment loop after waiting for 2s and tries to subscribe again. It fails now and continues after waiting for 10s (this keeps going on and on)
4. After (3), I turned the node on again, and the next attempt after 10s worked and started the subscription again 🔁 
5. After (3), I cancelled the context while it was waiting for 10s and it ended the goroutine due to switch case

Reply:
@im-adithya awesome!